// react-adminlte-dash/dist-modules/components/Navbar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _NavbarLogo = require('./NavbarLogo');

var _NavbarLogo2 = _interopRequireDefault(_NavbarLogo);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Navbar = function Navbar(_ref) {
  var children = _ref.children,
      logo = _ref.logo,
      sidebarToggle = _ref.sidebarToggle;
  return _react2.default.createElement(
    'header',
    { className: 'main-header' },
    logo,
    _react2.default.createElement(
      'nav',
      { className: 'navbar navbar-static-top' },
      _react2.default.createElement(
        'button',
        {
          className: 'sidebar-toggle',
          style: { border: 'none', outline: 'none' },
          onClick: function onClick() {
            return sidebarToggle();
          }
        },
        _react2.default.createElement(
          'span',
          { className: 'sr-only' },
          'Toggle navigation'
        )
      ),
      _react2.default.createElement(
        'div',
        { className: 'navbar-custom-menu' },
        _react2.default.createElement(
          'ul',
          { className: 'nav navbar-nav' },
          children
        )
      )
    )
  );
};

process.env.NODE_ENV !== "production" ? Navbar.propTypes = {
  children: _react2.default.PropTypes.node,
  logo: _react2.default.PropTypes.element,
  sidebarToggle: _react2.default.PropTypes.func.isRequired
} : void 0;

Navbar.defaultProps = {
  logo: _react2.default.createElement(_NavbarLogo2.default, null)
};

exports.default = Navbar;
