// react-adminlte-dash/dist-modules/components/NavbarLogo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var NavbarLogo = function NavbarLogo(_ref) {
  var link = _ref.link,
      logoLg = _ref.logoLg,
      logoSm = _ref.logoSm;
  return _react2.default.createElement(
    "a",
    {
      className: "logo",
      href: link
    },
    _react2.default.createElement(
      "div",
      { className: "logo-mini" },
      logoSm
    ),
    _react2.default.createElement(
      "div",
      { className: "logo-lg" },
      logoLg
    )
  );
};

process.env.NODE_ENV !== "production" ? NavbarLogo.propTypes = {
  link: _react2.default.PropTypes.string,
  logoLg: _react2.default.PropTypes.element,
  logoSm: _react2.default.PropTypes.element
} : void 0;

NavbarLogo.defaultProps = {
  link: '/',
  logoLg: _react2.default.createElement(
    "span",
    null,
    _react2.default.createElement(
      "b",
      null,
      "Admin"
    ),
    "LTE"
  ),
  logoSm: _react2.default.createElement(
    "span",
    null,
    _react2.default.createElement(
      "b",
      null,
      "A"
    ),
    "LT"
  )
};

exports.default = NavbarLogo;
