// react-adminlte-dash/dist-modules/components/SidebarMenuUser.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// User Status Options
// online: text-success
// offline: text-danger
// unknown: text-muted

var SidebarMenuUser = function SidebarMenuUser(props) {
  var userName = props.userName,
      userImage = props.userImage,
      showStatus = props.showStatus,
      userStatus = props.userStatus,
      userLink = props.userLink;

  // Set default image if no user image available

  var user = _react2.default.createElement('i', { className: 'fa fa-user fa-3x' });
  var statusLink = null;

  if (userImage) {
    user = _react2.default.createElement('img', {
      alt: 'User',
      className: 'img-circle',
      src: userImage
    });
  }

  // Deal with user status
  if (showStatus) {
    switch (userStatus.toLowerCase()) {
      case 'online':
        statusLink = _react2.default.createElement(
          'a',
          { href: userLink },
          _react2.default.createElement('i', { className: (0, _classnames2.default)('fa', 'fa-circle', 'text-success') }),
          'Online'
        );
        break;
      case 'offline':
        statusLink = _react2.default.createElement(
          'a',
          { href: userLink },
          _react2.default.createElement('i', { className: (0, _classnames2.default)('fa', 'fa-circle', 'text-danger') }),
          'Offline'
        );
        break;
      default:
        statusLink = _react2.default.createElement(
          'a',
          { href: userLink },
          _react2.default.createElement('i', { className: (0, _classnames2.default)('fa', 'fa-circle', 'text-muted') }),
          'Unknown'
        );
        break;
    }
  }

  return _react2.default.createElement(
    'div',
    { className: 'user-panel' },
    _react2.default.createElement(
      'div',
      { className: 'pull-left image' },
      user
    ),
    _react2.default.createElement(
      'div',
      { className: 'pull-left info' },
      _react2.default.createElement(
        'p',
        null,
        userName
      ),
      statusLink
    )
  );
};

process.env.NODE_ENV !== "production" ? SidebarMenuUser.propTypes = {
  userName: _react2.default.PropTypes.string,
  userImage: _react2.default.PropTypes.string,
  showStatus: _react2.default.PropTypes.bool,
  userStatus: _react2.default.PropTypes.string,
  userLink: _react2.default.PropTypes.string
} : void 0;

SidebarMenuUser.defaultProps = {
  userName: 'User Name',
  showStatus: true,
  userStatus: null,
  userLink: '/'
};

exports.default = SidebarMenuUser;
