// react-adminlte-dash/dist-modules/components/SidebarMenuSearch.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var SidebarMenuSearch = function (_React$Component) {
  _inherits(SidebarMenuSearch, _React$Component);

  function SidebarMenuSearch(props) {
    _classCallCheck(this, SidebarMenuSearch);

    var _this = _possibleConstructorReturn(this, (SidebarMenuSearch.__proto__ || Object.getPrototypeOf(SidebarMenuSearch)).call(this, props));

    _this.state = {
      value: _this.props.initialValue
    };
    _this._onSubmit = _this.onSubmit.bind(_this);
    _this._onInput = _this.onInput.bind(_this);
    return _this;
  }

  _createClass(SidebarMenuSearch, [{
    key: "onInput",
    value: function onInput(e) {
      this.setState({
        value: e.target.value
      });
    }
  }, {
    key: "onSubmit",
    value: function onSubmit(e) {
      e.preventDefault();
      this.props.onSearch(this.state.value);
    }
  }, {
    key: "render",
    value: function render() {
      return _react2.default.createElement(
        "form",
        {
          className: "sidebar-form",
          method: "get",
          onSubmit: this._onSubmit
        },
        _react2.default.createElement(
          "div",
          { className: "input-group" },
          _react2.default.createElement("input", {
            className: "form-control",
            name: "q",
            onChange: this._onInput,
            placeholder: "Search...",
            type: "text"
          }),
          _react2.default.createElement(
            "span",
            { className: "input-group-btn" },
            _react2.default.createElement(
              "button",
              {
                className: "btn btn-flat",
                id: "search-btn",
                name: "search",
                type: "submit"
              },
              _react2.default.createElement("i", { className: "fa fa-search" })
            )
          )
        )
      );
    }
  }]);

  return SidebarMenuSearch;
}(_react2.default.Component);

process.env.NODE_ENV !== "production" ? SidebarMenuSearch.propTypes = {
  initialValue: _react2.default.PropTypes.string,
  onSearch: _react2.default.PropTypes.func.isRequired
} : void 0;

SidebarMenuSearch.defaultProps = {
  initialValue: ''
};

exports.default = SidebarMenuSearch;
